sambacc/samba_cmds.py

```diff
@@ -172,7 +172,10 @@ def winbindd_foreground() -> SambaCommand:
 
 
 def samba_dc_foreground() -> SambaCommand:
-    return samba_dc["--foreground"]
+    samba_foreground = samba_dc["--foreground"]
+    if _DAEMON_CLI_STDOUT_OPT in get_samba_specifics():
+        samba_foreground = samba_foreground["--debug-stdout"]
+    return samba_foreground
```

Comment:
Do you foresee any problem with the following suggestion(similar to _smbd_ and _winbindd_)?

```suggestion
    return samba_dc["--foreground", _daemon_stdout_opt("samba")]
```

Reply:
without the SAMBA_SPECIFICS=daemon_cli_debug_output you will get error `Invalid option --stdout: unknown option`

Reply:
@anoopcs9 it works with the `SAMBA_SPECIFICS` variable set and have `daemon_cli_debug_output` value.
_daemon_stdout_opt("samba") function return `--stdout` if the `SAMBA_SPECIFICS` variable is not set or has a value other than `daemon_cli_debug_output`. The Samba binary has no `--stdout` arguments and fails.

```py
def _daemon_stdout_opt(daemon: str) -> str:
    if daemon == "smbd":
        opt = "--log-stdout"
    else:
        opt = "--stdout"
    opt_lst = get_samba_specifics()
    if _DAEMON_CLI_STDOUT_OPT in opt_lst:
        opt = "--debug-stdout"
    return opt
```

Reply:
Just now realized that corresponding [Containerfile](https://github.com/samba-in-kubernetes/samba-container/blob/master/images/ad-server/Containerfile.centos) for ad-server fails to update `SAMBA_SPECIFICS` during build process.

So the right solution would be to use _daemon_stdout_opt()_ here and subsequently fix the build.

Reply:
something like this?

Reply:
I don't think we have to modify _daemon_stdout_opt()_. Just the change from my earlier [suggestion](https://github.com/samba-in-kubernetes/sambacc/pull/106#discussion_r1477726791) is enough. In addition `samba-ad-server` image build process can be updated to set `SAMBA_SPECIFICS` from [samba-container](https://github.com/samba-in-kubernetes/samba-container) repository. Ultimately that's the container image in use here.

Reply:
> In addition `samba-ad-server` image build process can be updated to set `SAMBA_SPECIFICS` from [samba-container](https://github.com/samba-in-kubernetes/samba-container) repository.

https://github.com/samba-in-kubernetes/samba-container/commit/6100b0f49f9dcedaa5da58c85cb49b2c29f9ca20